### PR TITLE
faster direction for covidcast

### PR DESCRIPTION
**Pull request by dfarrow0**
avoid quadratic time at the database by scanning entire time-series in linear time in python

### CODE REVIEW

**Comment by dfarrow0** [comment]
@capnrefsmmat please take a look

File: src/acquisition/covidcast/direction.py [hunk 1]
```diff
@@ -50,4 +50,39 @@ def get_direction(x, y, n=1):
       return 0
     else:
       # return an integer in {-1, +1}
-      return int(numpy.sign(fit.slope))
+      return int(np.sign(fit.slope))
+
+  @staticmethod
+  def scan_timeseries(offsets, row_days, values, timestamp1s, timestamp2s):
```
Comment (dfarrow0) [inline src/acquisition/covidcast/direction.py:56]: @capnrefsmmat i assume you care most about this part

**Comment by dfarrow0** [comment]
back from the future to report the results: 502573 directions computed in 5:05 minutes (down from 3-4 hours to do roughly the same).

as a quick sanity check, all 59434 directions called on days 20200417+ were the same as what was previously computed using the slow method.

i still owe unit tests.

**Comment by dfarrow0** [comment]
@krivard thanks for taking a look and for the comment

**Comment by capnrefsmmat** [comment]
Logic looks good to me. I also don't see a way to simplify the logic any, but it's already quite clear and simple. And 1,700 directions calculated per second isn't so bad.

**Pull request merged**